src/main/java/io/cryostat/recordings/RecordingsModule.java

```diff
@@ -148,4 +157,29 @@ static ScheduledExecutorService provideNotificationScheduler() {
         Runtime.getRuntime().addShutdownHook(new Thread(ses::shutdown));
         return ses;
     }
+
+    @Provides
+    // FIXME which path to use?
```

Comment:
Add a `@Singleton` here. Without that, Dagger will inject a new manager instance anywhere you've requested it. This means each of your API handlers has its own manager instance, with its own in-memory labels map - which is why when you POST to create a recording the response indicates the labels were captured, but when you GET after they appear to be missing.

Reply:
Thanks! I'll let you know when I'm ready for another review.